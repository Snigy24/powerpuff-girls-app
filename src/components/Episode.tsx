import React from "react";
import { EpisodeDetails } from "../Types/ListEpisodeType";

const Episode = (props: EpisodeDetails) => {
  return (
    <div>
      <p>{props.episodeName}</p>
      <img src={props.coverImage} alt="Powerpuff girl" />
      <p>{props.summary}</p>
      <p>{props.releaseDate}</p>
    </div>
  );
};

export default Episode;
