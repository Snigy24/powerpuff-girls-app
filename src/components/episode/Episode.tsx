import { EpisodeDetails } from "../../Types/ListEpisodeType";

/**
 * The function takes in an object of type EpisodeDetails and returns a div with the episodeName,
 * coverImage, summary, and releaseDate.
 * @param {EpisodeDetails} props - EpisodeDetails
 * @returns A React component that displays the episode details.
 */
const Episode = (props: EpisodeDetails) => {
  return (
    <div>
      <p>{props.episodeName}</p>
      <img src={props.coverImage} alt="Powerpuff girl" />
      <p>{props.summary}</p>
      <p>{props.releaseDate}</p>
    </div>
  );
};

export default Episode;
