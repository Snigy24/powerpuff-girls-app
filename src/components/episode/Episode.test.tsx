import renderer from "react-test-renderer";
import Episode from "./Episode";

/* Creating a fake object to pass into the Episode component. */
const fakeEpisodeDetails = {
  episodeName: "Fake Name",
  coverImage: "Fake Image",
  summary: "Fake Description",
  releaseDate: "2014",
};

it("renders correctly", () => {
  const tree = renderer.create(<Episode {...fakeEpisodeDetails} />).toJSON();
  expect(tree).toMatchSnapshot();
});
