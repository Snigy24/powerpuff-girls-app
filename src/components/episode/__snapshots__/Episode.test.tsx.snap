// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Fake Name
  </p>
  <img
    alt="Powerpuff girl"
    src="Fake Image"
  />
  <p>
    Fake Description
  </p>
  <p>
    2014
  </p>
</div>
`;
